public/js/visualisation_helpers.js

```diff
@@ -0,0 +1,65 @@
+function VisualisationHelpers() {
+}
+
+VisualisationHelpers.prototype._get_colors_for_evalue = function (evalue,hits) {
+  var colors = d3.scale
+      .log()
+      .domain([
+        d3.min([1e-5, d3.min(hits.map(function (d) {
+          if (parseFloat(d.evalue) === 0.0) return undefined;
+          return d.evalue;
+        }))]),
+        d3.max(hits.map(function (d) {
+          return d.evalue;
+        }))
+      ])
+      .range([40,150]);
+  var rgb = colors(evalue);
+  return d3.rgb(rgb,rgb,rgb);
+}
+
+// Returns copy of `arr` containing only unique values. Assumes duplicate
+// values always occur consecutively (which they will if `arr` is sorted).
+VisualisationHelpers.prototype._uniq = function (arr) {
+  var uniq = [];
+  for(var i = 0; i < arr.length; i++) {
+    while(i < (arr.length - 1) && arr[i] === arr[i + 1]) { i++; }
+    uniq.push(arr[i]);
+  }
+  return uniq;
+}
+
+/**
+ * Defines how ticks will be formatted.
+ *
+ * Modified by Priyam based on https://github.com/mbostock/d3/issues/1722.
```

Comment:
Now that it's out of kablammo folder, better to say - "borrowed from kablammo - modified by ..."